packages/ai-code-completion/src/common/code-completion-agent.ts

```diff
@@ -36,82 +36,89 @@ export class CodeCompletionAgentImpl implements CodeCompletionAgent {
         context: monaco.languages.InlineCompletionContext,
         token: monaco.CancellationToken
     ): Promise<monaco.languages.InlineCompletions | undefined> {
-        const languageModel =
-            await this.languageModelRegistry.selectLanguageModel({
-                agent: this.id,
-                ...this.languageModelRequirements[0],
+        const progress = await this.progressService.showProgress(
```

Comment:
I was wondering why we are not using the `withProgress` but just noticed that this method always provides something cancellable and I guess this is not something we want in this case.

I'm also wondering whether it makes sense to report progress on the different steps (create prompt, send request, formulate response). However, while adding transparency, it might be too fine-grained.

Reply:
Everything except the LLM request is <1ms, you will never see it atm